llm/tools/ipfs.ts

```diff
@@ -0,0 +1,45 @@
+import { PinataSDK } from "pinata";
+import type { UploadResponse } from "pinata";
+import { DynamicStructuredTool } from "@langchain/core/tools";
+import { z } from "zod";
+import "dotenv/config";
+
+const pinata = new PinataSDK({
+  pinataJwt: process.env.PINATA_JWT,
+  pinataGateway: process.env.GATEWAY_URL,
+});
+
+export async function upload_to_ipfs(name: string, content: object): Promise<UploadResponse> {
+  try {
+    const file = new File([JSON.stringify(content)], name, {
+      type: "text/json",
+    });
+    const upload = await pinata.upload.public.file(file);
+    console.log(upload);
+    return upload;
+
+  } catch (error) {
+    console.error("Error uploading to IPFS:", error);
+    throw error;
+  }
+}
+
+export const uploadHeroTool = new DynamicStructuredTool({
```

Comment:
This is great!

Can you add another tool in similiar style for `Entity`? These are the enemies and bosses that will be spawned for the round tasks (door, dragon, knight, angry mob etc)

Reply:
After the discussion on slack, I guess I'll keep the Hero metadata upload for reference, but won't add this. Agreed?